docs/resources/restriction_policy.md

```diff
@@ -0,0 +1,56 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "datadog_restriction_policy Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Provides a Datadog RestrictionPolicy resource. This can be used to create and manage Datadog restriction_policy.
+---
+
+# datadog_restriction_policy (Resource)
+
+Provides a Datadog RestrictionPolicy resource. This can be used to create and manage Datadog restriction_policy.
+
+## Example Usage
+
+```terraform
+# Create new restriction_policy resource
+
+
+resource "datadog_restriction_policy" "foo" {
+  bindings {
+    principals = ["role:00000000-0000-1111-0000-000000000000"]
+    relation   = "editor"
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `resource_id` (String) Identifier for the resource, formatted as type:id.
+
+### Optional
+
+- `bindings` (Block List) (see [below for nested schema](#nestedblock--bindings))
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--bindings"></a>
+### Nested Schema for `bindings`
+
+Optional:
+
+- `principals` (List of String) An array of principals. A principal is a subject or group of subjects. Each principal is formatted as `type:id`. Supported types: `role` and `org`. The org ID can be obtained through the api/v2/current_user API.
```

Comment:
since we are only mentioning role and org, do you think we would need a FF in the backend to block teams/users temporarily intil that feature is GA'ed?

Reply:
not gating teams, but not documenting yet so we'll leave this as is